### DISCUSSION TITLE
CircleCI pipeline fails to build AMI image [ERROR: error: unbalanced parenthesis at position 230]

### DISCUSSION THREAD

**Discussion by nadiia-kotelnikova** [gruntwork-io/knowledge-base#664] [Help] [answered]
Hello all,

i am trying to build AMI with CircleCI pipelines. I took the examples of CircleCI scripts from [here](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/_ci/scripts). The AMI build script I took from [here](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/shared/us-west-2/_regional/amis). 

The CircleCI gives the following error, looks like from [this line](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/examples/for-production/infrastructure-live/shared/us-west-2/_regional/amis/build_eks_cluster_instance.sh#L105):
```
#!/bin/bash -eo pipefail
./_ci/scripts/run-build-scripts.sh "$SOURCE_REF" "$CIRCLE_BRANCH"
The following build scripts were updated:
shared-services/eu-central-1/_global/amis/build_eks_cluster_instance.sh
Calling build script shared-services/eu-central-1/_global/amis/build_eks_cluster_instance.sh
2023-02-09 13:07:30 [INFO] [aws-auth] Assuming role arn:aws:iam::502221110000:role/allow-auto-deploy-from-other-accounts. These creds will expire after 3600 seconds.
2023-02-09 13:07:32 [INFO] [aws-auth] Success!
[infrastructure-deployer] INFO[2023-02-09T13:07:33Z] Invoking Lambda function ecs-deploy-runner-invoker to trigger deployment. 
ERROR: error: unbalanced parenthesis at position 230
```

I assume that the bash line was somehow not correctly parsed by `ecs-deploy-runner-invoker` Lambda function. 

Have someone saw something like this? Is there any workaround?

---

<ins datetime="2023-02-09T13:20:02Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109887">Tracked in ticket #109887</a></p>
</ins>

**Comment by zackproser** [comment]
Hi @nadiia-caspar, 

1. Could you please tell me what  you're trying to accomplish? What is your goal?
2. What documentation are you following?
3. Did your company already have a Reference Architecture deployed?

Thanks!

**Reply by nadiia-kotelnikova** [reply]
Hi @zackproser 
1. We recently added CI/CD CircleCI pipelines, so we don't have to deploy changes from a local machine. My goal now is to upload or update a `build_...sh` scripts for AMIs and I expect that images will be built by ECS AMI builder as part of CircleCI pipelines. So the logs which I shared earlier are from the CircleCI scripts.

2. I took the examples of CircleCI scripts from [here](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/_ci/scripts). The AMI build script I took from [here](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/shared/us-west-2/_regional/amis).

3. Yes, the reference architecture is already deployed. And most of the modules are from service-catalog or cis-service-catalog repos.

**Reply by nadiia-kotelnikova** [reply]
@zackproser do you maybe have any ideas about my problem?

**Comment by nadiia-kotelnikova** [comment, marked as answer]
Hello, grunties!

I figured out the problem. I had a typo in `allowed_repos_regex` value, so the `--packer-template-path` option for the `infrastructure-deployer` command could not be compared correctly with allowed repos.

**Comment by zackproser** [comment]
@nadiia-caspar Roger - thank you for closing the loop here. I really appreciate it.
